tests/src/CompilerTest.php

```diff
@@ -732,11 +742,28 @@ public function testFailParameterPropertyPath()
             'a' => 'foo'
         );
 
-        $mysql = null;
-        $phpInput = null;
-        $phpOutput = null;
-
-        $this->verify($scenario, $method, $arguments, $mysql, $phpInput, $phpOutput);
+        // verify the exception
+        $this->verifyException(
+            $scenario,
+            $method,
+            $arguments,
+            Compiler::ERROR_BAD_FILTER_EXPRESSION,
+            array(
+                'class' => 'Person',
+                'table' => 0,
+                'arguments' => array(
+                    3,
+                    'match',
+                    array(
+                        5,
+                        array(2, 'name'),
+                        array(1, 'a'),
+                        array(2, 'first')
+                    ),
+                    array(1, 'regex')
+                )
+            )
```

Comment:
We could improve the readability if we were to give this value a variable name. That way, the input and expected output would all be together and labeled, and the "verifyException" calls would all look the same.

That would minimize the cognition burden on the reader. That way, the reader wouldn't have to keep tracking of the nesting levels, or to try to figure out what the value means.